lib/oban/query.ex

```diff
@@ -254,6 +255,34 @@ defmodule Oban.Query do
 
   defp unique_query(_changeset), do: nil
 
+  defp unique_field({changeset, :args, [_ | _] = keys}, acc) do
+    args =
+      changeset
+      |> Changeset.get_field(:args)
+      |> Map.new(fn {key, val} -> {to_string(key), val} end)
+      |> Map.take(keys)
+
+    dynamic([j], fragment("? @> ?", j.args, ^args) and ^acc)
+  end
+
+  defp unique_field({changeset, :args, _}, acc) do
+    args = Changeset.get_field(changeset, :args)
+
+    dynamic([j], j.args == ^args and ^acc)
+  end
+
+  defp unique_field({changeset, :queue, _}, acc) do
+    queue = Changeset.get_field(changeset, :queue)
+
+    dynamic([j], j.queue == ^queue and ^acc)
+  end
+
+  defp unique_field({changeset, :worker, _}, acc) do
+    worker = Changeset.get_field(changeset, :worker)
+
+    dynamic([j], j.worker == ^worker and ^acc)
+  end
```

Comment:
as an alternative to these 3 definitions
```
  defp unique_field({changeset, field, _}, acc) when field in [:args, :queue, :worker] do
    value = Changeset.get_field(changeset, field)

    dynamic([j], field(j, ^field) == ^value and ^acc)
  end
```

Reply:
Yes! The `field/2` syntax is what I was looking for earlier. Updated without the field guard, those are already validated by this point.